tools/shell/tests/conftest.py

```diff
@@ -0,0 +1,175 @@
+import pytest
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--shell-binary", action="store", default=None, help="Provide the shell binary to use for the tests"
+    )
+    parser.addoption("--start-offset", action="store", type=int, help="Skip the first 'n' tests")
+
+
+def pytest_collection_modifyitems(config, items):
+    start_offset = config.getoption("--start-offset")
+    if not start_offset:
+        # --skiplist not given in cli, therefore move on
+        return
+
+    skipped = pytest.mark.skip(reason="included in --skiplist")
+    skipped_items = items[:start_offset]
+    for item in skipped_items:
+        item.add_marker(skipped)
+
+
+import subprocess
+import sys
+from typing import List
+
+
+class ShellTest:
+    def __init__(self, shell):
+        if not shell:
+            raise ValueError("Please provide a shell binary")
+        self.shell = shell
+        self.arguments = [shell, '--batch', '--init', '/dev/null']
+        self.statements: List[str] = []
+        self.input = None
+        self.output = None
+
+    def add_argument(self, arg):
+        self.arguments.append(arg)
+        return self
+
+    def statement(self, stmt):
+        self.statements.append(stmt)
+        return self
```

Comment:
We call this enough times for every test, probably better to switch it to being a varargs (splat) method

Reply:
I tried that, didn't like it
This has the same functionality with less `,` hassle